sdk/containerregistry/container-registry/src/blob/containerRegistryBlobClient.ts

```diff
@@ -407,15 +408,33 @@ export class ContainerRegistryBlobClient {
       "ContainerRegistryBlobClient.downloadBlob",
       options,
       async (updatedOptions) => {
-        const { readableStreamBody } = await this.client.containerRegistryBlob.getBlob(
+        const initialResponse = await this.client.containerRegistryBlob.getBlob(
           this.repositoryName,
           digest,
           updatedOptions
         );
 
+        assertHasProperty(initialResponse, "readableStreamBody");
+        assertHasProperty(initialResponse, "contentLength");
+
         return {
           digest,
-          content: readableStreamBody ?? Readable.from([]),
+          content: new RetriableReadableStream(
```

Comment:
Are you able to progressively validate the digest as the caller reads the stream?

Reply:
We can definitely do that. I implemented it a while ago on another branch, actually, but never got around to making a PR for it! I would like to keep the digest validation in a separate stream wrapping the retriable stream, for separation of concern and so that if we ever get around to sharing this code with Storage, we'd be able to drop in the Storage version without any changes.

Reply:
Sounds good!

In .NET, I added a separate optional lambda, but I 100% defer to you on what's best in TS!  Mine also hasn't yet been through the rigor of a Core code review, so we'll see where it lands :)

Reply:
After looking into it, I changed to a callback-based approach like .NET does. Using `pipe` to chain streams together, like I was doing, means that errors won't be handled properly.